sql/handler.cc

```diff
@@ -5948,6 +5948,9 @@ int handler::ha_update_row(const uchar *old_data, uchar *new_data)
   DBUG_ASSERT(new_data == table->record[0]);
   DBUG_ASSERT(old_data == table->record[1]);
 
+  // InnoDB changes sys_trx_end to curr_trx_id and we need to restore MAX_TRX
+  if (table->versioned() && !table->versioned_by_sql())
```

Comment:
1. Use `if (table->file->check_table_binlog_row_based(1))` instead, don't use `versioned()` condition.
2. rename `binlog_log_row_internal` to `binlog_log_row`, delete old `binlog_log_row`.
3. add `table->file->check_table_binlog_row_based(1)` check to all places (3) where `binlog_log_row` used:
```c++
  if (likely(!error) && table->file->check_table_binlog_row_based(1))
  {
    rows_changed++;
    error= binlog_log_row(table, buf, 0, log_func);
  }
```